dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/message/MessageImpl.java

```diff
@@ -478,27 +483,232 @@ public Object put(String key, Object value) {
+        }
+        
+    }
+    
+    final class KeySet extends AbstractSet<String> {
+        public final int size() {
+            return this.size();
+        }
+        public final void clear() {
+            this.clear();
+        }
+        public final Iterator<String> iterator() {
+            return new KeyIterator(); 
+        }
+        public final boolean contains(Object o) {
+            return containsKey(o);
+        }
+        public final boolean remove(Object key) {
+            if (containsKey(key)) {
+                this.remove(key);
+                return true;
+            }
+            return false;
+        }
+        public final Spliterator<String> spliterator() {
+           throw new UnsupportedOperationException();
+        }
+        public final void forEach(Consumer<? super String> action) {
+            if (action == null) {
+                throw new NullPointerException();
+            }
+            for (int i = 0; i < TOTAL; i++) {
+                if (propertyValues[i] != NOT_FOUND) {
+                    action.accept(propertyNames[i]);
+                }
+            }
+            for (String k : backedKeySet()) {
+                action.accept(k);
+            }
+        }
+    }
+    
+    final class Values extends AbstractCollection<Object> {
+        public final int size() {
+            return this.size();
+        }
+        public final void clear() {
+            this.clear();
+        }
+        public final Iterator<Object> iterator() {
+            return new ValuesIterator(); 
+        }
+        public final boolean contains(Object o) {
+            return containsValue(o);
+        }
+        public final Spliterator<Object> spliterator() {
+           throw new UnsupportedOperationException();
+        }
+        public final void forEach(Consumer<? super Object> action) {
+            if (action == null) {
+                throw new NullPointerException();
+            }
+            for (int i = 0; i < TOTAL; i++) {
+                if (propertyValues[i] != NOT_FOUND) {
+                    action.accept(propertyValues[i]);
+                }
+            }
+            for (Object v : backedValues()) {
+                action.accept(v);
             }
         }
-        return myEntrySet;
     }
     
+    final class EntrySet extends AbstractSet<Map.Entry<String, Object>> {
+        public final int size() {
+            return this.size();
+        }
+        public final void clear() {
+            this.clear();
+        }
+        public final Iterator<Map.Entry<String, Object>> iterator() {
+            return new EntryIterator(); 
+        }
+        public final boolean contains(Object o) {
+            if (!(o instanceof Map.Entry)) {
+                return false;
+            }
+            Map.Entry<?, ?> e = (Map.Entry<?, ?>) o;
+            String key = (String) e.getKey();
+            
+            if (containsKey(key)) {
```

Comment:
Can you just use getOrDefault here like in other places instead of containsKey and get?  Will save doing a scan again